File: Mini_Games/Maths_Quiz/Maths_Quiz.py
import random

Player = False

while Player == False:
    # Begin/Reset round
    user_option = input("What questions would you like? \nAddition \nSubtraction \nDivision \nMultiplication\n")
    
    # Allow user to select their range
    num_start = int(input("Start range: "))
    num_end = int(input("End range: "))

    number_one = random.randrange(num_start, num_end)
    number_two = random.randrange(num_start, num_end)

    # Questions + Answers
    if user_option == "Addition":
        problem = f"{number_one} + {number_two} = "
        solution = number_one + number_two
    elif user_option == "Subtraction":
        problem = f"{number_one} - {number_two} = "
        solution = number_one - number_two
    elif user_option == "Division":
        problem = f"{number_one} / {number_two} = "
        solution = number_one / number_two
    elif user_option == "Multiplication":
        problem = f"{number_one} * {number_two} = "
        solution = number_one * number_two
    else:
        print("Invalid option. Please choose Addition, Subtraction, Division, or Multiplication.")
        continue  # Restart the loop if the option is invalid

    # Scoring
    user_answer = input(problem)
    if float(user_answer) == solution:  
        print("Correct!")
    else: 
        print("Nice try!, but the correct answer is " + str(solution))

    # Section to start/end game
    play = input("\nPlay again? Yes or No: ").lower()
    if play == "yes":
        Player = False
    else:
        print("Thanks for playing!")
        break

File: Games_Website/all-js/Maths_quiz.js
let numberOne;
let numberTwo;
let solution;
let gameSection = document.getElementById("MathsQ-game-section");
let problemText = document.getElementById("problem");
let feedback = document.getElementById("feedback");
let submitButton = document.getElementById("submit-button");
let playAgainButton = document.getElementById("MathsQ-play-again");
let tryAgainButton = document.getElementById("MathsQ-try-again");
let exitButton = document.getElementById("MathsQ-exit");
let userAnswerInput = document.getElementById("user-answer");

function startGame() {
    let startRange = parseInt(document.getElementById("MathsQ-start-range").value);
    let endRange = parseInt(document.getElementById("MathsQ-end-range").value);
    let questionType = document.getElementById("MathsQ-question-type").value;

    if (isNaN(startRange) || isNaN(endRange) || startRange >= endRange) {
        alert("Please enter valid numbers for the range.");
        return;
    }

    numberOne = Math.floor(Math.random() * (endRange - startRange + 1)) + startRange;
    numberTwo = Math.floor(Math.random() * (endRange - startRange + 1)) + startRange;

    switch (questionType) {
        case "Addition":
            problemText.textContent = `${numberOne} + ${numberTwo} = `;
            solution = numberOne + numberTwo;
            break;
        case "Subtraction":
            problemText.textContent = `${numberOne} - ${numberTwo} = `;
            solution = numberOne - numberTwo;
            break;
        case "Division":
            problemText.textContent = `${numberOne} / ${numberTwo} = `;
            solution = numberOne / numberTwo;
            break;
        case "Multiplication":
            problemText.textContent = `${numberOne} * ${numberTwo} = `;
            solution = numberOne * numberTwo;
            break;
        default:
            alert("Invalid option. Please choose Addition, Subtraction, Division, or Multiplication.");
            return;
    }

    gameSection.style.display = "block";
    playAgainButton.style.display = "none";
    tryAgainButton.style.display = "none";
    exitButton.style.display = "none";
    feedback.textContent = "";
    submitButton.style.display = "inline";
    userAnswerInput.disabled = false;
    userAnswerInput.value = "";
}

function submitAnswer() {
    let userAnswer = parseFloat(userAnswerInput.value);

    if (isNaN(userAnswer)) {
        feedback.textContent = "Invalid input. Please enter a valid number.";
        return;
    }

    if (userAnswer === solution) {
        feedback.textContent = "Correct!";
        submitButton.style.display = "none";
        playAgainButton.style.display = "block";
        userAnswerInput.disabled = true;
    } else {
        feedback.textContent = `Incorrect, try again or exit?`;
        tryAgainButton.style.display = "block";
        exitButton.style.display = "block";
        userAnswerInput.disabled = true;
    }
}

function tryAgain() {
    userAnswerInput.value = "";
    feedback.textContent = "";
    tryAgainButton.style.display = "none";
    exitButton.style.display = "none";
    userAnswerInput.disabled = false;
}

function restartGame() {
    document.getElementById("MathsQ-start-range").value = "";
    document.getElementById("MathsQ-end-range").value = "";
    userAnswerInput.value = "";
    gameSection.style.display = "none";
    playAgainButton.style.display = "none";
    feedback.textContent = "";
    tryAgainButton.style.display = "none";
    exitButton.style.display = "none";
    submitButton.style.display = "inline";
    userAnswerInput.disabled = false;
}
